app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/AcceptedOwnedDetailViewModel.java

```diff
@@ -43,13 +48,24 @@ private void fetchRequest(String bookId) {
                 .addOnFailureListener(e -> errorMessage.setValue(SCAN_ERROR.UNEXPECTED));
     }
 
-    public void handleScannedIsbn(String currentIsbn, String scannedIsbn) {
+    public void handleScannedIsbn(Book currentBook, String currentIsbn, String scannedIsbn) {
         if (currentIsbn.equals(scannedIsbn)) {
-            requestRepository.updateRequestStatus(request.getValue(), REQUEST_STATUS.HANDING)
-                    .addOnSuccessListener(updatedRequest -> request.setValue(updatedRequest))
-                    .addOnFailureListener(e -> errorMessage.setValue(SCAN_ERROR.UNEXPECTED));
-        } else {
-            errorMessage.setValue(SCAN_ERROR.INVALID_ISBN);
+            if (request.getValue().getStatus().equals(REQUEST_STATUS.ACCEPTED)) {
+                requestRepository.updateRequestStatus(request.getValue(), REQUEST_STATUS.HANDING)
+                        .addOnSuccessListener(updatedRequest -> request.setValue(updatedRequest))
+                        .addOnFailureListener(e -> errorMessage.setValue(SCAN_ERROR.UNEXPECTED));
+            } else if (request.getValue().getStatus().equals(REQUEST_STATUS.RETURNING)) {
+                requestRepository.updateRequestStatus(request.getValue(), REQUEST_STATUS.OPENED)
```

Comment:
the request here should be deleted instead, `REQUEST_STATUS.OPENED` means the book is having a 'REQUESTED' status